File: Programmers/Level1/콜라츠추측.py
def solution(num):
    answer = 0
    cnt = 0
    while 1:
        if cnt >= 500:
            return -1
        if num == 1:
            return cnt
        if num%2:
            num = num*3 + 1
        else:
            num //= 2
        
        cnt += 1

num = 16
print(solution(num))

File: Programmers/Level1/콜라츠추측/콜라츠추측.js
function solution(num) {
  let cnt = 0;

  while (true) {
    if (cnt >= 500) {
      return -1;
    } else if (num === 1) {
      return cnt;
    } else if (num % 2) {
      num = num * 3 + 1;
    } else {
      num = parseInt(num / 2);
    }
    cnt += 1;
  }
}

const num = 6;

console.log(solution(num));
